src/common/mipmap_cache.c

```diff
@@ -839,16 +888,34 @@ static void _init_f(float *out, uint32_t *width, uint32_t *height, const uint32_
     // demosaic during downsample
     if(image->filters != 9u)
     {
+      const int method = dt_image_get_demosaic_method(imgid);
+
       // Bayer
       if(image->bpp == sizeof(float))
       {
-        dt_iop_clip_and_zoom_demosaic_half_size_crop_blacks_f(out, (const float *)buf.buf, &roi_out, &roi_in,
-                                                              roi_out.width, roi_in.width, image, 1.0f);
+        if(method == 3)
```

Comment:
Just comparing against a constant seems like a bad idea, but i also did not like the idea of "find out what value DT_IOP_DEMOSAIC_PASSTHROUGH_MONOCHROME".
So i came up with this version.